tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -1683,6 +1691,28 @@ port::StatusOr<DeviceMemory<uint8>> CreateRnnWorkspace(
   return workspace_allocator->AllocateBytes(workspace_size_in_bytes);
 }
 
+port::Status CopyRnnSeqLengthsH2D(
+    Stream* stream, const CudnnRnnSequenceTensorDescriptor& input_desc,
+    DeviceMemory<uint8>* dev_seq_lengths) {
+  std::vector<int> seq_lengths(input_desc.batch_size());
+  cudnnDataType_t data_type;
+  cudnnRNNDataLayout_t data_layout;
+  int max_seq_length, batch_size, vector_size;
+  float padding_fill;
+	RETURN_IF_CUDNN_ERROR(cudnnGetRNNDataDescriptor(
+      /*RNNDataDesc=*/input_desc.data_handle(), /*dataType=*/&data_type,
+      /*layout=*/&data_layout, /*maxSeqLength=*/&max_seq_length,
+      /*batchSize=*/&batch_size, /*vectorSize=*/&vector_size,
+      /*arrayLengthRequested=*/input_desc.batch_size(),
+      /*seqLengthArray=*/&seq_lengths[0],
+      /*paddingFill=*/(void*)&padding_fill));
+  uint8* begin_ptr = reinterpret_cast<uint8*>(seq_lengths.data());
+  std::vector<uint8> seq_lengths_buf(
+      begin_ptr, begin_ptr + seq_lengths.size() * sizeof(int));
+  stream->ThenMemcpyH2D<uint8>(seq_lengths_buf, dev_seq_lengths);
```

Comment:
Just so I understand, you're relying on this memcpy being synchronous (otherwise you'd somehow have to keep `seq_lengths_buf` alive until the memcpy completed)?

Reply:
You're right, and this is also my concern. I think for the small seqLengthsArray, the CUDA can make sure the memcpy is doing what it is expected. However, it might be risky when it is a large array.
 Alternatively, I am thinking we can keep the seqLengthsArray on device by removing the `.HostMemory("sequence_lengths")` for `CudnnRNNV3`. And use ThenMemcpyD2H (if we have this function) to copy back to host a seqLengthsArray so as to create the input tensor descriptor.

Reply:
Or, is there any method to keep two copies of sequence_lengths that one is on device and the other is on host?

Reply:
Figured out a new way to place the seq_lengths array into the rnn_seq_tensor object to make sure it can live long enough during the computation.